llvm/lib/Target/RISCV/RISCVFrameLowering.cpp

```diff
@@ -1847,11 +1847,16 @@ bool RISCVFrameLowering::assignCalleeSavedSpillSlots(
       MFI.setStackID(FrameIdx, TargetStackID::ScalableVector);
   }
 
-  // Allocate a fixed object that covers the full push or libcall size.
   if (RVFI->isPushable(MF)) {
-    if (int64_t PushSize = RVFI->getRVPushStackSize())
-      MFI.CreateFixedSpillStackObject(PushSize, -PushSize);
+    // Allocate a fixed object that covers all the registers that are pushed.
```

Comment:
I'm a bit curious if this object is needed at all.

Reply:
We might not have created an object for each register. See `void @spill_x10()` in `llvm/test/CodeGen/RISCV/push-pop-popret.ll`, where from what I can tell, we only have an object for `s10` and `s11`. For the same reasons as with save/restore, we cannot clobber the stack used by the other (lower-numbered) callee-saved registers, because we definitely restore into them with `pop`

Reply:
ok thanks.